sr_utilities_common/src/sr_utilities_common/aws_manager.py

```diff
@@ -61,13 +61,26 @@ def __init__(self):
             aws_session_token=aws_session_token
         )
 
-    def get_bucket_structure_with_prefix(self, bucket_name, prefix):
-        if prefix:
-            try:
-                if 'Contents' in self._client.list_objects(Bucket=bucket_name, Prefix=prefix):
-                    return self._client.list_objects(Bucket=bucket_name, Prefix=prefix)['Contents']
-            except self.client.exceptions.NoSuchBucket as e:
-                rospy.logerr("Failed listing bucket objects. " + str(e))
+    def __init__(self, access_key, secret_key, session_token):
+        self.file_full_paths = []
+        self.aws_paths = []
+        self._client = boto3.client(
+            's3',
+            aws_access_key_id=access_key,
+            aws_secret_access_key=secret_key,
+            aws_session_token=session_token
+        )
+
+    def get_bucket_structure_with_prefix(self, bucket_name, prefix=""):
+        try:
+            if prefix:
+                command = self._client.list_objects(Bucket=bucket_name, Prefix=prefix)
+            else:
+                command = self._client.list_objects(Bucket=bucket_name)
```

Comment:
command = self._client.list_objects(Bucket=bucket_name)
if prefix:
                command = self._client.list_objects(Bucket=bucket_name, Prefix=prefix)
```suggestion
            if prefix:
                command = self._client.list_objects(Bucket=bucket_name, Prefix=prefix)
            else:
                command = self._client.list_objects(Bucket=bucket_name)
```